File: Data Structures & Algorithms/partition-equal-subset-sum/submission-6.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        totalSum = sum(nums)

        if totalSum %2 != 0:
            return False
        
        halfSum = totalSum // 2
        
        dp = set()
        dp.add(0)

        for i in range(len(nums) -1, -1, -1):
            nextDP = set()
            for t in dp:
                if (t + nums[i]) == halfSum:
                    return True
                nextDP.add(t) #Skip
                nextDP.add(t + nums[i]) #include
            dp = nextDP

        return True if halfSum in dp else False

File: Data Structures & Algorithms/missing-number/submission-6.java
class Solution {
    public int missingNumber(int[] nums) {
        int missing_num = nums.length;
        for (int i = 0; i < nums.length; i++){
            missing_num += i;
            missing_num -= nums[i];
        }
        return missing_num;
    }
}
